state-sync/src/request_manager.rs

```diff
@@ -480,13 +484,15 @@ impl RequestManager {
         let multicast_start_time = self.get_multicast_start_time(version).unwrap_or(UNIX_EPOCH);
         if is_timeout(multicast_start_time, self.multicast_timeout) {
             let new_multicast_level = std::cmp::min(
-                self.multicast_level
-                    .checked_add(1)
-                    .ok_or_else(|| format_err!("New multicast level has overflown!"))?,
+                self.multicast_level.checked_add(1).ok_or_else(|| {
+                    Error::IntegerOverflow("New multicast level has overflown!".into())
+                })?,
                 self.upstream_config
                     .upstream_count()
                     .checked_sub(1)
-                    .ok_or_else(|| format_err!("Upstream count has overflown!"))?, // multicast_level (=network preference) is 0-indexed
+                    .ok_or_else(|| {
+                        Error::IntegerOverflow("Upstream count has overflown!".into())
+                    })?, // multicast_level (=network preference) is 0-indexed
             );
```

Comment:
Lol looking at code like this is just like what?  I wish error handling of checked stuff could be easier.

Could you move these blobs out of the `min` comparison?

```
let multicast = multicast_level + 1; // Paraphrasing
let upstream = upstream_count - 1; // Paraphrasing
let new_level = min(multicast, upstream);
```

Still, this blob is confusing even just looking at that what the levels and stuff are, you'll have to tell me later.

Reply:
I've moved it out. Perhaps we can revisit it -- I'm sure there's other places in the code that are just as unclear 😄